File: sdks/local-rp/java/src/main/java/community/catalyst/linkkeys/localrp/RevocationError.java
package community.catalyst.linkkeys.localrp;

/** A sibling-signed revocation certificate failed to meet quorum. Mirrors {@code liblinkkeys::revocation::RevocationError}. */
public class RevocationError extends RuntimeException {
    public RevocationError(int got, int need) {
        super("revocation certificate has " + got + " valid distinct signer(s), needs " + need);
    }
}

File: sdks/local-rp/csharp/src/LinkKeys.LocalRp/RevocationError.cs
namespace LinkKeys.LocalRp;

/// <summary>A sibling-signed revocation certificate failed to meet quorum. Mirrors <c>liblinkkeys::revocation::RevocationError</c>.</summary>
public class RevocationError(int got, int need) : Exception($"revocation certificate has {got} valid distinct signer(s), needs {need}");
